Add Unit tests for taxonomies

joemcgill opened the pull request:
### Issue Number
Related to #81.
~Blocked by #94.~

### Description
This adds several unit tests methods related to taxonomies.

- Adds method `test_get_url_list_custom_taxonomy()` that tests that custom taxonomies are returned by `Core_Sitemaps_Taxonomies::get_url_list()`
- Adds method `test_get_url_list_custom_taxonomy_private()` that tests that private custom taxonomies are _not_ returned by `Core_Sitemaps_Taxonomies::get_url_list()`
- Fixes a bug which allowed `Core_Sitemaps_Taxonomies::get_url_list()` to return non-public data
- Adds method `test_get_url_list_taxonomies()` which ensures that default taxonomy sitemaps include expected URL lists.
- Adds method `test_get_sitemap_entries_custom_taxonomies()`, which assures public custom taxonomies are included in the sitemap index and that private custom taxonomies are not.

### Screenshots (before and after if applicable)
If your PR includes visual changes include before and after screenshots showing the change.

### Type of change
Please select the relevant options:

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Enhancement (change which improves an existing feature. E.g., performance improvement, docs update, etc.)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Acceptance criteria
- [x] My code follows WordPress coding standards.
- [x] I have performed a self-review of my own code.
- [ ] If the changes are visual, I have cross browser / device tested.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generate no new warnings.
- [ ] I have added test instructions that prove my fix is effective or that my feature works.

Commit:
Add Unit tests for custom taxonomies sitemap lists.

- Adds method `test_get_url_list_custom_taxonomy()` that tests that custom taxonomies are returned by `Core_Sitemaps_Taxonomies::get_url_list()`
- Adds method `test_get_url_list_custom_taxonomy_private()` that tests that private custom taxonomies are _not_ returned by `Core_Sitemaps_Taxonomies::get_url_list()`
- Fixes a bug which allowed `Core_Sitemaps_Taxonomies::get_url_list()` to return non-public data

Commit:
Merge branch 'feature/81-unit-test-providers' into feature/81-unit-test-providers-taxonomies

Commit:
Return full objects from Core_Sitemaps_Taxonomies::get_object_sub_types()

This reverts a change made in 7573025 which retured a single dimentional array of taxonomy names instead of full taxonomy objects and updates the logic in the `get_url_list()` method so tests still pass.

Commit:
Coding standards cleanup

Commit:
Add unit test for default taxonomy URL lists

- Adds method `test_get_url_list_taxonomies()` which ensures that default taxonomy sitemaps include expected URL lists.

Commit:
Add unit test for custom taxonomies in the sitemap index

This adds test method, `test_get_sitemap_entries_custom_taxonomies()`, which assures public custom taxonomies are included in the sitemap index and that private custom taxonomies are not.

Commit:
Remove unnecessary string concatenation

swissspidy commented on tests/phpunit/class-test-core-sitemaps.php:
```diff
@@ -321,6 +331,40 @@ public function test_get_sitemap_entries_custom_post_types() {
+		$private_term = $this->factory->term->create( array( 'taxonomy'  => 'private_taxonomy' ) );
+
+		// Create a test post applied to all test terms.
+		$this->factory->post->create_and_get(
```
```suggestion
		self::factory()->post->create_and_get(
```

swissspidy commented on tests/phpunit/class-test-core-sitemaps.php:
```diff
@@ -321,6 +331,40 @@ public function test_get_sitemap_entries_custom_post_types() {
+
+		// Create test terms in the custom taxonomy.
+		$public_term  = $this->factory->term->create( array( 'taxonomy'  => 'public_taxonomy' ) );
+		$private_term = $this->factory->term->create( array( 'taxonomy'  => 'private_taxonomy' ) );
```
```suggestion
		$private_term = self::factory()->term->create( array( 'taxonomy'  => 'private_taxonomy' ) );
```

swissspidy commented on tests/phpunit/class-test-core-sitemaps.php:
```diff
@@ -321,6 +331,40 @@ public function test_get_sitemap_entries_custom_post_types() {
+		register_taxonomy( 'private_taxonomy', 'post', array( 'public' => false ) );
+
+		// Create test terms in the custom taxonomy.
+		$public_term  = $this->factory->term->create( array( 'taxonomy'  => 'public_taxonomy' ) );
```
```suggestion
		$public_term  = self::factory()->term->create( array( 'taxonomy'  => 'public_taxonomy' ) );
```

swissspidy commented on tests/phpunit/class-test-core-sitemaps.php:
```diff
@@ -422,6 +466,117 @@ public function test_get_url_list_cpt_private() {
+		$categories = array_merge( array( 1 ), self::$cats );
+
+		// Create a test post to calculate update times.
+		$post = $this->factory->post->create_and_get(
```
```suggestion
		$post = self::factory()->post->create_and_get(
```

swissspidy commented on tests/phpunit/class-test-core-sitemaps.php:
```diff
@@ -422,6 +466,117 @@ public function test_get_url_list_cpt_private() {
+		register_taxonomy( $taxonomy, 'post' );
+
+		// Create test terms in the custom taxonomy.
+		$terms = $this->factory->term->create_many( 10, array( 'taxonomy'  => $taxonomy ) );
```
```suggestion
		$terms = self::factory()->term->create_many( 10, array( 'taxonomy'  => $taxonomy ) );
```

swissspidy commented on tests/phpunit/class-test-core-sitemaps.php:
```diff
@@ -422,6 +466,117 @@ public function test_get_url_list_cpt_private() {
+		$terms = $this->factory->term->create_many( 10, array( 'taxonomy'  => $taxonomy ) );
+
+		// Create a test post applied to all test terms.
+		$post = $this->factory->post->create_and_get( array( 'tax_input' => array( $taxonomy => $terms ) ) );
```
```suggestion
		$post = self::factory()->post->create_and_get( array( 'tax_input' => array( $taxonomy => $terms ) ) );
```

swissspidy commented on tests/phpunit/class-test-core-sitemaps.php:
```diff
@@ -422,6 +466,117 @@ public function test_get_url_list_cpt_private() {
+		register_taxonomy( $taxonomy, 'post', array( 'public' => false ) );
+
+		// Create test terms in the custom taxonomy.
+		$terms = $this->factory->term->create_many( 10, array( 'taxonomy'  => $taxonomy ) );
```
```suggestion
		$terms = self::factory()->term->create_many( 10, array( 'taxonomy'  => $taxonomy ) );
```

swissspidy commented on tests/phpunit/class-test-core-sitemaps.php:
```diff
@@ -422,6 +466,117 @@ public function test_get_url_list_cpt_private() {
+		$terms = $this->factory->term->create_many( 10, array( 'taxonomy'  => $taxonomy ) );
+
+		// Create a test post applied to all test terms.
+		$this->factory->post->create( array( 'tax_input' => array( $taxonomy => $terms ) ) );
```
```suggestion
		self::factory()->post->create( array( 'tax_input' => array( $taxonomy => $terms ) ) );
```

swissspidy committed:
Access factory via static method

See https://core.trac.wordpress.org/ticket/33968 for reference.

Commit:
Move clean up routines before assertions.

Commit:
Merge branch 'feature/81-unit-test-providers-taxonomies' of github.com:GoogleChromeLabs/wp-sitemaps into feature/81-unit-test-providers-taxonomies

Commit:
Merge branch 'master' into feature/81-unit-test-providers-taxonomies

joemcgill commented:
@swissspidy I've updated this one to point to `master` now that #94 is merged and moved clean up routines to before assertions. Let me know if there is anything else to do here.
